crates/uv-distribution/src/metadata/requires_dist.rs

```diff
@@ -222,6 +213,92 @@ fn apply_source_strategy(
     }
 }
 
+/// Resolve the dependency groups (which may contain references to other groups) into concrete
+/// lists of requirements.
+fn resolve_dependency_groups(
+    groups: &BTreeMap<&GroupName, &Vec<DependencyGroupSpecifier>>,
+) -> BTreeMap<GroupName, Result<Vec<uv_pep508::Requirement<VerbatimParsedUrl>>, MetadataError>> {
```

Comment:
It's unclear to me if we want to track the _original_ group of requirements throughout the resolver for error messages.